File: examples/FSharp/Logging/ElasticSearchLogging.fs
module FSharp.HttpTests.ElasticSearchLogging

open System.Threading.Tasks
open FSharp.Control.Tasks.V2.ContextInsensitive

open NBomber.Contracts
open NBomber.FSharp

// in this example we use:
// - Serilog.Sinks.Elasticsearch (https://github.com/serilog/serilog-sinks-elasticsearch)
// to get more info about logging, please visit: (https://nbomber.com/docs/logging)

let run () =

    let step = Step.create("step_1", fun context -> task {

        do! Task.Delay(seconds 1)

        // this message will be saved to elastic search
        context.Logger.Debug("hello from NBomber")

        return Response.Ok()
    })

    Scenario.create "hello_world_scenario" [step]
    |> Scenario.withoutWarmUp
    |> Scenario.withLoadSimulations [KeepConstant(copies = 1, during = seconds 30)]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.withTestSuite "logging"
    |> NBomberRunner.withTestName "elastic_search"
    |> NBomberRunner.loadInfraConfig "./Logging/infra-config.json"
    |> NBomberRunner.run
    |> ignore

File: examples/CSharp/Logging/ElasticSearchLogging.cs
using System;
using System.Threading.Tasks;

using NBomber.Contracts;
using NBomber.CSharp;

namespace CSharp.Logging
{
    // in this example we use:
    // - Serilog.Sinks.Elasticsearch (https://github.com/serilog/serilog-sinks-elasticsearch)
    // to get more info about logging, please visit: (https://nbomber.com/docs/logging)

    public class ElasticSearchLogging
    {
        public static void Run()
        {
            var step = Step.Create("step", async context =>
            {
                await Task.Delay(TimeSpan.FromSeconds(1));

                // this message will be saved to elastic search
                context.Logger.Debug("hello from NBomber");

                return Response.Ok();
            });

            var scenario = ScenarioBuilder
                .CreateScenario("scenario", step)
                .WithoutWarmUp()
                .WithLoadSimulations(new[]
                {
                    Simulation.KeepConstant(copies: 1, during: TimeSpan.FromSeconds(30))
                });

            NBomberRunner
                .RegisterScenarios(scenario)
                .WithTestSuite("logging")
                .WithTestName("elastic_search")
                .LoadInfraConfig("./Logging/infra-config.json")
                .Run();
        }
    }
}
